core-library/src/main/java/org/silverpeas/core/contribution/content/wysiwyg/service/WysiwygController.java

```diff
@@ -367,4 +378,55 @@ public static void indexEmbeddedLinkedFiles(FullIndexEntry indexEntry,
   public static String createPath(String componentId, String context) {
     return getManager().createPath(componentId, context);
   }
+
+  private static LocalizedContribution contributionFrom(final String componentId,
+      final String objectId, final String language) {
+    return contributionFrom(componentId, objectId, language, true);
+  }
+
+  private static LocalizedContribution contributionFrom(final String componentId,
+      final String objectId, final String language, boolean indexable) {
+    return new LocalizedContribution() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 34 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1188)